### DISCUSSION TITLE
Omit specific fields from the generated type

### DISCUSSION THREAD

**Discussion by iSplasher** [MichalLytek/typegraphql-prisma#83] [Q&A] [answered]
Hello, first of all, awesome library!

I want to ask if there's an approach to what I want to achieve.
Let's say I have a model like this
```
model AttributeTemplate {
    id Int @id @default(autoincrement())

    name String
    type AttributeTemplateType

    attributes Json

    by      User @relation(fields: [user_id], references: [id])
    user_id Int

    created_at DateTime @default(now())
    updated_at DateTime @updatedAt
}
```

To retrieve many of these items that are only owned by the user, I have a custom resolver that is using the generated types like this
```
  @Query((returns) => [AttributeTemplate])
  async templates(
    @Args() args: FindManyAttributeTemplateArgs,
    @Ctx() ctx: AuthContext
  ) {
    const items = await state.service
      .get(ServiceType.Template)
      .getMany(args, ctx.user);
    return items;
  }
```

This works well, I make sure to check for the correct user in the Template service to ensure only owned objects are returned. Now, the only thing I want is to omit the `by` field from the generated type of the model so that this ownership is happening transparently since I'm already getting the user from the context. This is even more so in the case of a mutation. There the `by` field is required, that's why I want to hide it from the GraphQL type if possible.

I have tried all kinds of weird hacks like this to try and hide it but nothing works
```
@InputType()
class Test extends FindManyAttributeTemplateArgs;
delete Object.getPrototypeOf(Test )["by"];
```

The only thing that works is redefining the type
```
type AttributeTemplateCreateInputT = Omit<
  AttributeTemplateCreateInput,
  'by' | 'attributes'
>;

@InputType()
export class AttributeTemplateCreateInputC
  implements AttributeTemplateCreateInputT {
  @Field((type) => AttributeTemplateType)
  type!: AttributeTemplateCreateInput['type'];
}
```

But having to redefine all types is not ideal. That's why I want to ask for input.
I want to remove the `by` field from all InputTypes.

**Comment by MichalLytek** [comment, marked as answer]
@iSplasher Have you tried reading the docs?
https://github.com/MichalLytek/typegraphql-prisma/blob/main/Readme.md#hiding-prisma-model-field-in-graphql-schema

**Reply by MichalLytek** [reply]
For now it's not supported to omit model field in input type. You need to modify the generated code by yourself.

**Reply by iSplasher** [reply]
Hello @MichalLytek 
You're right, I can't believe I missed that in the docs!
I will be eagerly waiting for the `input: true` option to be implemented.
I might be able to submit a pull request if you can point me in the right direction.

**Reply by iSplasher** [reply]
https://github.com/MichalLytek/typegraphql-prisma/blob/d2c2eb20e6dd6b02c43f723f8ea9621c7ae0ac7f/src/generator/type-class.ts#L144

Would a check for the omitted fields here suffice?

**Reply by MichalLytek** [reply]
It's not so simple - please be patient and wait for a proper implementation. We need to check all the related inputs and outputs as well (like aggregate).
